astroid/inference.py

```diff
@@ -174,10 +174,11 @@ def _infer_map(
             values = _update_with_replacement(values, unpack_items)
         else:
             key = helpers.safe_infer(name, context=context)
-            value = helpers.safe_infer(value, context=context)
-            if any(not elem for elem in (key, value)):
+            safe_value = helpers.safe_infer(value, context=context)
+            if any(not elem for elem in (key, safe_value)):
```

Comment:
L178 makes sure that `safe_value` on L181 is only `type[NodeNG] | type[Proxy]` as `Uninferable` is also filtered here.

This is crucial as that means that `Uninferable` will never end up in `Dict.items` as an `InferenceError` will be raised.

Sorry for the churn, I didn't know `Uninferable` did that and didn't recognise `mypy` warning us about it.